airqo-platform/AirQo-api
# Airqo frequency measurements

# Average Airqo measurements by frequency

**_WHAT DOES THIS PR DO?_**
Computes the average values for device measurements based on frequency then sends the computed values to device registry for storage.

Frequencies considered :
```
minute
hourly
daily
weekly
monthly
```
**_WHAT JIRA STORY/TASK IS RELATED TO THIS PR?_**
None

**_WHAT IS THE LINK TO THE PR BRANCH_**
[airqo-frequency-measurements](https://github.com/airqo-platform/AirQo-api/tree/ft-airqo-frequency-measurements)

**_HOW DO I TEST OUT FUNCTIONALITY?_**
Checkout this branch

```
cd AirQo-api/src/data-mgt/python/jobs/airqo-frequency-measurements
python3 -m venv venv
source venv/bin/activate
pip install -r requirements.txt
export FREQUENCY=hourly
export DEVICE_REGISTRY_URL=http://staging-platform.airqo.net/api/v1/
python calculate_measurements_by_frequency.py
```

**_HOW DO I RUN TESTS?_**
Checkout this branch

```
cd AirQo-api/src/data-mgt/python/jobs/airqo-frequency-measurements
python3 -m venv venv
source venv/bin/activate
pip install -r requirements.txt
python python frequency_tests.py
```
You should see averaged measurements in the database that have a frequency 

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 
None

**_IS THERE ANY JENKINS CONSOLE LINK FOR CODE COVERAGE AND BUILD INFO?_**
None

**_ARE THERE ANY RELATED PRs?_**
#363
